src/extra/Markdown/index.tsx

```diff
@@ -138,11 +139,111 @@ type components = {
 	[element: string]: (props: any) => JSX.Element | React.ReactElement | Element;
 };
 
+interface Decorator {
+	match: RegExp;
+	component: string;
+}
+
+interface DecoratorPluginOptions {
+	decorators: Decorator[];
+}
+
+const decoratorPlugin = (options: DecoratorPluginOptions) => {
+	options = options || { decorators: [] };
+
+	if (!options.decorators || !options.decorators.length) {
+		return;
+	}
+
+	const targets: Array<{ decorator: Decorator; node: any }> = [];
+
+	const visitor = (node: any) => {
+		const matchedDecorator = options.decorators.find((decorator) => {
+			return (
+				node.children &&
+				node.children.length === 1 &&
+				node.children[0].type === 'text' &&
+				decorator.match.test(node.children[0].value) &&
+				node.children[0]
+			);
+		});
+
+		if (matchedDecorator) {
+			targets.push({
+				decorator: matchedDecorator,
+				node,
+			});
+		}
+	};
+
+	const modify = () => {
+		targets.forEach((target) => {
+			const node = target.node;
+			const match = target.decorator.match;
+
+			const text = node.children[0].value;
+			const children = [];
+
+			match.lastIndex = 0;
+			do {
+				const lastIndex = match.lastIndex;
+				const result = match.exec(text);
+
+				if (result) {
+					children.push(
+						{
+							type: 'element',
+							tagName: 'span',
+							children: [
+								{
+									type: 'text',
+									value: text.substring(lastIndex, result.index),
+								},
+							],
+						},
+						{
+							type: 'element',
+							tagName: target.decorator.component,
```

Comment:
IMO it would be better to have span as default with a class, with some pre-defined styles.

Reply:
I don't see how having span as default component would benefit. And we should avoid using plain classes in favor of styled-components.

Reply:
Did you test this? I am not sure it would work that way

Reply:
@iamsolankiamit yes, I've manually tested it. Can you share your concerns? You can try yourself by passing these parameters to the markdown component:

```typescript
const decorators: Decorator[] = [{
  match: new RegExp(`((@{1,2}|#|!{1,2})[a-z\\d-_\\/]+(\\.[a-z\\d-_\\/]+)*)`, 'gmi'),
  component: 'mention'
}];

const componentOverrides = {
  mention: (props: any) => {
    return <Txt.span {...props} color="red" />
  }
};
```

Reply:
Ahh didn't realise we have componentOverrides. We should document this and have a default mention component.

Reply:
This pr is not about exactly mentions, it just unlocks mentions. Mentions could be added in separate pr. Now this is enough for me to unblock JF bug.